pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -110,23 +150,29 @@ func haveOverlap(a1, a2 []string) bool {
 // It evaluates if a pod can fit due to the volumes it requests, and those that
 // are already mounted. If there is already a volume mounted on that node, another pod that uses the same volume
 // can't be scheduled there.
-// This is GCE, Amazon EBS, ISCSI and Ceph RBD specific for now:
+// This is GCE, Amazon EBS, Ceph RBD, ISCSI, and PVC specific for now:
 // - GCE PD allows multiple mounts as long as they're all read-only
 // - AWS EBS forbids any two pods mounting the same volume ID
 // - Ceph RBD forbids if any two pods share at least same monitor, and match pool and image, and the image is read-only
 // - ISCSI forbids if any two pods share at least same IQN and ISCSI volume is read-only
+// - PVC forbids if more than one pod shares a PVC with the same name and the PVC uses the ReadWriteOncePod access mode
 func (pl *VolumeRestrictions) Filter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
 	for i := range pod.Spec.Volumes {
 		v := &pod.Spec.Volumes[i]
 		// fast path if there is no conflict checking targets.
-		if v.GCEPersistentDisk == nil && v.AWSElasticBlockStore == nil && v.RBD == nil && v.ISCSI == nil {
+		if v.GCEPersistentDisk == nil && v.AWSElasticBlockStore == nil && v.RBD == nil && v.ISCSI == nil && v.PersistentVolumeClaim == nil {
 			continue
 		}
 
 		for _, ev := range nodeInfo.Pods {
 			if isVolumeConflict(v, ev.Pod) {
 				return framework.NewStatus(framework.Unschedulable, ErrReasonDiskConflict)
 			}
+
+			isAccessModeConflict, code, reason := pl.isAccessModeConflict(v, ev.Pod)
```

Comment:
I'm not sure if this is the right plugin to do this check. Did you consider volumebinding plugin?

@chendave WDYT? Also, were we checking the other modes before? I couldn't find any usages in scheduler code.

Reply:
That plugin could likely also contain this logic if it is a better place. I thought volume restrictions made sense due of [this description of the plugin](https://github.com/kubernetes/kubernetes/blob/v1.21.1/pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go#L108-L110):
```
// It evaluates if a pod can fit due to the volumes it requests, and those that
// are already mounted. If there is already a volume mounted on that node, another pod that uses the same volume
// can't be scheduled there.
```

In this case, a pod would not be schedulable due to the volumes already mounted on that node.

Reply:
Actually, it was @cofyc who wrote the volumebinding plugin. Perhaps you have some idea of the questions above?